web/app/js/components/Navigation.jsx

```diff
@@ -0,0 +1,329 @@
+    },
+  },
+  toolbar: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'flex-end',
+    padding: '0 8px',
+    ...theme.mixins.toolbar,
+  },
+  content: {
+    flexGrow: 1,
+    width: `calc(100% - ${drawerWidth}px)`,
+    backgroundColor: theme.palette.background.default,
+    padding: theme.spacing.unit * 3,
+  },
+  linkerdLogoContainer: {
+    backgroundColor: theme.palette.primary.dark,
+  },
+  linkerdNavLogo: {
+    minWidth: "180px",
+  },
+  navMenuItem: {
+    paddingLeft: "24px",
+    paddingRight: "24px",
+  },
+});
+
+class NavigationBase extends React.Component {
+  constructor(props) {
+    super(props);
+    this.api = this.props.api;
+    this.handleApiError = this.handleApiError.bind(this);
+
+    this.state = this.getInitialState();
+  }
+
+  getInitialState() {
+    return {
+      drawerOpen: true,
+      resourceMenuOpen: false,
+      helpMenuOpen: false,
+      latestVersion: '',
+      isLatest: true,
+      namespaceFilter: "all"
+    };
+  }
+
+  componentDidMount() {
+    this.fetchVersion();
+  }
+
+  fetchVersion() {
+    let versionUrl = `https://versioncheck.linkerd.io/version.json?version=${this.props.releaseVersion}&uuid=${this.props.uuid}&source=web`;
+    this.versionPromise = fetch(versionUrl, { credentials: 'include' })
+      .then(rsp => rsp.json())
+      .then(versionRsp => {
+        let latestVersion;
+        let parts = this.props.releaseVersion.split("-", 2);
+        if (parts.length === 2) {
+          latestVersion = versionRsp[parts[0]];
+        }
+        this.setState({
+          latestVersion,
+          isLatest: latestVersion === this.props.releaseVersion
+        });
+      }).catch(this.handleApiError);
+  }
+
+  handleApiError(e) {
+    this.setState({
+      error: e
+    });
+  }
+
+  handleDrawerOpen = () => {
+    this.setState({ drawerOpen: true });
+  };
+
+  handleDrawerClose = () => {
+    this.setState({ drawerOpen: false });
+  };
+
+  handleResourceMenuClick = () => {
+    this.setState(state => ({ resourceMenuOpen: !state.resourceMenuOpen }));
+  };
+
+  handleHelpMenuClick = () => {
+    this.setState(state => ({ helpMenuOpen: !state.helpMenuOpen }));
+  }
+
+  menuItem(path, title, icon) {
+    const { classes, api } = this.props;
+    let normalizedPath = this.props.location.pathname.replace(this.props.pathPrefix, "");
+    let isCurrentPage = path => path === normalizedPath;
+
+    return (
+      <MenuItem
+        component={Link}
+        to={api.prefixLink(path)}
+        className={classes.navMenuItem}
+        selected={isCurrentPage(path)}>
+        <ListItemIcon>{icon}</ListItemIcon>
+        <ListItemText primary={title} />
+      </MenuItem>
+    );
+  }
+  render() {
+    const { classes, ChildComponent } = this.props;
+
+    return (
+      <div className={classes.root}>
+        <AppBar
+          position="absolute"
+          className={classNames(classes.appBar, this.state.drawerOpen && classes.appBarShift)}>
```

Comment:
More of a readability nit: I am not entirely sure how `classNames` interprets statements like `this.state.drawerOpen && classes.appBarShift` but I found it a little difficult to comprehend. I tried rewriting the statement like so:
```jsx
className={classNames(classes.appBar, {[`${classes.appBarShift}`]: this.state.drawerOpen})}>
```
And that seemed to work the same way.

Reply:
Hahahah, what that line is actually doing is a combination of cool es6 tricks:

First, it's using the `&&` operand to return the string name of the class if `drawerOpen` is true.
(The js `&&` operand will return the latter of the provided values if the first expression is truthy). Example:
```
>> false && "foo"
>> false

>> true && "foo"
>> "foo"
```

Then, it's using the fancy object assignment to assign `appBarShift`'s className to the object if provided:
```
>> let a = "a_string";
>> let obj = { a }

>> obj
{a: "a_string"}
```

So this is equivalent to providing an object whose key is `classes.appBarShift` and value is `this.state.drawerOpen` as you've written. I'll rewrite it to be less trick-y!

Reply:
Oh my gosh! so much sorcery! That is pretty cool. it's a TIOLI so feel free to do whatever you think is best.

Reply:
Hahaha, yeah I think readable > cool, so I'll change them to more traditional args. 
It's a pattern I see popping up a lot in the material examples, so maybe we can change later.